packages/access/src/accesscontrol/accesscontrol.cairo

```diff
@@ -227,13 +308,35 @@ pub mod AccessControlComponent {
         fn _grant_role(
             ref self: ComponentState<TContractState>, role: felt252, account: ContractAddress,
         ) {
-            if !AccessControl::has_role(@self, role, account) {
-                let caller: ContractAddress = get_caller_address();
-                self.AccessControl_role_member.write((role, account), true);
+            if !self.is_role_granted(role, account) {
+                let caller = starknet::get_caller_address();
+                let role_info = AccountRoleInfo { is_active: true, effective_from: 0 };
+                self.AccessControl_role_member.write((role, account), role_info);
                 self.emit(RoleGranted { role, account, sender: caller });
             }
         }
 
+        /// Attempts to grant `role` to `account` that will become effective when `delay` passes.
+        ///
+        /// Internal function without access restriction.
+        ///
+        /// May emit a `RoleGrantedWithDelay` event.
+        fn _grant_role_with_delay(
+            ref self: ComponentState<TContractState>,
+            role: felt252,
+            account: ContractAddress,
+            delay: u64,
+        ) {
+            assert(delay > 0, Errors::INVALID_DELAY);
+            if !self.is_role_granted(role, account) {
```

Comment:
Since now users may want to use the grant_role_function to update a delay, or the grant_role function to remove the delay, these functions silently passing may confuse them. What do you think about adding an error on the else condition stating that the user already has the role?